kafka-rest.openapi.yaml

```diff
@@ -0,0 +1,4545 @@
+      in: query
+      required: true
+      schema:
+        $ref: '#/components/schemas/AclPatternType'
+
+    AclPermission:
+      name: 'permission'
+      description: 'The ACL permission.'
+      in: query
+      required: false
+      schema:
+        $ref: '#/components/schemas/AclPermission'
+
+    AclPermissionRequired:
+      name: 'permission'
+      description: 'The ACL permission.'
+      in: query
+      required: true
+      schema:
+        $ref: '#/components/schemas/AclPermission'
+
+    AclPrincipal:
+      name: 'principal'
+      description: 'The ACL principal.'
+      in: query
+      required: false
+      schema:
+        type: string
+
+    AclResourceName:
+      name: 'resource_name'
+      description: 'The ACL resource name.'
+      in: query
+      required: false
+      schema:
+        type: string
+
+    AclResourceType:
+      name: 'resource_type'
+      description: 'The ACL resource type.'
+      in: query
+      required: false
+      schema:
+        $ref: '#/components/schemas/AclResourceType'
+
+    AclResourceTypeRequired:
+      name: 'resource_type'
+      description: 'The ACL resource type.'
+      in: query
+      required: true
+      schema:
+        $ref: '#/components/schemas/AclResourceType'
+
+    BrokerId:
+      name: 'broker_id'
+      description: 'The broker ID.'
+      in: path
+      required: true
+      schema:
+        type: integer
+      example: 1
+
+    ClusterId:
+      name: 'cluster_id'
+      description: 'The Kafka cluster ID.'
+      in: path
+      required: true
+      schema:
+        type: string
+      example: 'cluster-1'
+
+    ConfigName:
+      name: 'name'
+      description: 'The configuration parameter name.'
+      in: path
+      required: true
+      schema:
+        type: string
+      example: 'compression.type'
+
+    ConsumerGroupId:
+      name: 'consumer_group_id'
+      description: 'The consumer group ID.'
+      in: path
+      required: true
+      schema:
+        type: string
+      example: 'consumer-group-1'
+
+    ConsumerId:
+      name: 'consumer_id'
+      description: 'The consumer ID.'
+      in: path
+      required: true
+      schema:
+        type: string
+      example: 'consumer-1'
+
+    IncludeAuthorizedOperations:
+      name: 'include_authorized_operations'
+      description: 'Specify if authorized operations should be included in the response.'
+      in: query
+      required: false
+      schema:
+        type: boolean
+
+    # NOTE: Kafka-rest's OpenAPI spec has a known undocumented inconsistency where
+    #       the list topics operation expects the includeAuthorizedOperations query parameter
+    #       whereas the get topic operation expects the include_authorized_operations query parameter.
+    #       We declare both here in our version of the spec to at least ensure that it is at least documented.
+    #       The inconsistency has been set in stone for a while and is unlikely to change in the near future.
```

Comment:
@jlrobins: I believe this comment here about Kafka REST's API spec inconsistency might strike a chord with you.